### PR TITLE
MAINT: remove python2/3 compatibility code from examples, closes #501

### LINKED ISSUE #501: Python 2/3 compability in examples
We currently have lots of examples where we do the standardized

```
# Imports for common Python 2/3 codebase
from __future__ import print_function, division, absolute_import
from future import standard_library
standard_library.install_aliases()
```

in the beginning. I personally feel that this is not good to do in general, mostly because it makes the examples technical and harder to read, as well as forcing users to scroll unnecessarily much to get to the point. My suggestion is to write rewrite the examples such that this is not necessary (where, if anywhere, this is needed) and then simply remove these lines.

### PR DESCRIPTION
**Pull request by adler-j**
Someone (@kohr-h) needs to run `py.test --examples` using python 3 to verify that this works as expected.

### CODE REVIEW

File: examples/space/simple_r.py [hunk 1]
```diff
@@ -45,32 +40,33 @@ def __eq__(self, other):
...
+        return RealNumber(self, value)
 
-    class Vector(odl.LinearSpaceVector):
-        """Real vectors are floats
-        """
 
-        __val__ = None
+class RealNumber(odl.LinearSpaceVector):
+    """Real vectors are floats
+    """
```
Comment (kohr-h) [inline examples/space/simple_r.py]: Dot and `"""` on the same line

File: examples/space/simple_rn.py [hunk 2]
```diff
@@ -38,7 +29,7 @@ class SimpleRn(odl.space.base_ntuples.FnBase):
     """The real space R^n, non-optimized implmentation."""
 
     def __init__(self, size):
-        super().__init__(size, np.float)
+        odl.space.base_ntuples.FnBase.__init__(self, size, np.float)
```
Comment (kohr-h) [inline examples/space/simple_rn.py]: Perhaps here a `from odl.space.base_ntuples import FnBase` would be more readable.

**Comment by adler-j** [comment]
Did you run the examples?

**Comment by kohr-h** [comment]
Everything works fine except the ones involving proximal and STIR, which I haven't installed. 
All good except for some minor stuff.

**Comment by adler-j** [comment]
merge after CI

**Pull request merged**